Simplenote/Classes/SPOptionsViewController.m

```diff
@@ -649,4 +708,33 @@ - (BOOL)themePref
     return [[NSUserDefaults standardUserDefaults] boolForKey:SPThemePref];
 }
 
+#pragma mark - Picker view delegate
+- (NSInteger)numberOfComponentsInPickerView:(UIPickerView *)pickerView {
+    return 1;
+}
+
+- (NSInteger)pickerView:(UIPickerView *)pickerView numberOfRowsInComponent:(NSInteger)component {
+    return [timeoutPickerOptions count];
+}
+
+- (void)pickerView:(UIPickerView *)pickerView didSelectRow:(NSInteger)row inComponent:(NSInteger)component {
+    [self.pinTimeoutTextField setText:timeoutPickerOptions[row]];
+    
+    [[NSUserDefaults standardUserDefaults] setInteger:row forKey:kPinTimeoutPreferencesKey];
+    [self.tableView reloadData];
+}
+
+- (void)pinTimeoutDoneAction:(id)sender
+{
+    [self.pinTimeoutTextField resignFirstResponder];
+}
+
+- (NSAttributedString *)pickerView:(UIPickerView *)pickerView attributedTitleForRow:(NSInteger)row forComponent:(NSInteger)component {
+    VSTheme *theme = [[VSThemeManager sharedManager] theme];
+    NSAttributedString *attributedTitle =
+    [[NSAttributedString alloc] initWithString:timeoutPickerOptions[row] attributes:@{NSForegroundColorAttributeName:[theme colorForKey:@"textColor"]}];
```

Comment:
Maybe put the allocation and the assignment in the same line? (and move the `attributes` to a variable to shorten the line length?)